tensorflow/stream_executor/rocm/rocm_dnn.cc

```diff
@@ -3711,8 +3711,6 @@ bool MIOpenSupport::DoBatchNormalizationForwardImpl(
 
   auto status = miopenStatusInvalidValue;
   if (is_training) {
-    stream->ThenMemZero(batch_mean, batch_mean->size());
-    stream->ThenMemZero(batch_var, batch_var->size());
```

Comment:
nevermind....read up on the code, and it seems `batch_mean` and `batch_var` are "inouts" (both input and output during training).

The inputs are a running mean/variance values for the previous batches, and outputs are the running mean/variance values updated for this batch.

Reply:
When exponential_average_factor is the default value (1), zeroing out is redundant, because their contents is replaced with new values.

When it is not 1, the MIOpen call calculates something like
```new_batch_mean = exp_avg_factor*current_batch_mean + (1-exp_avg_factor)*old_batch_mean```

where old_batch_mean comes from those tensors.

Zeroing out batch_mean before the call results in incorrect numbers.